File: 프로그래머스연습문제/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    int number = x;
    int divisor = 0;
    while(number >= 10){
        divisor += number%10;
        number /= 10;
    }
    divisor += number;
    
    if(x % divisor == 0){}
    else answer = false;

    return answer;
}

File: 프로그래머스_python/연습문제/Level1/하샤드 수.py
def solution(x):
    
    
    number = x
    total = 0
    
    while number > 0:
        total = total + number % 10
        number = number // 10
    
    if x % total == 0:
        answer = True
    else:
        answer = False
    
    return answer
